### PR TITLE
Part of #20506: Using vectors for EditorObjectiveOptions

### PR DESCRIPTION
**Pull request by aramk-hub**
As stated by @duncanspumpkin , we had two sources of truth for the checks here, and I was given the task of replacing them in this file with a vector and its size variable. If I misunderstood the function of it and it needs a revamp, do let me know (I am still finding my way around the codebase and the different objects and methods). It follows what the previous parts of this opened issue has done closely, so let me know if there are any issues.

### CODE REVIEW

File: src/openrct2-ui/windows/EditorObjectiveOptions.cpp [hunk 1]
```diff
@@ -134,6 +134,9 @@ static uint64_t window_editor_objective_options_page_hold_down_widgets[] = {
 
 class EditorObjectiveOptionsWindow final : public Window
 {
+private:
+    std::vector<RideId> _riddenRides;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/EditorObjectiveOptions.cpp:138]: These aren't ridden rides these are just a list of all rides rides (what a confusing term...). Could you rename it to rideable rides (rideable rides is also a rubbish name perhaps you can think of something better) and add a comment stating `// Not shops or facilities`.

**Comment by duncanspumpkin** [comment]
A little additional context about ride rides. In the codebase we have a structure called Ride that all rides, facilities (toilets) and shops use. This is a list of just the rides. Hence the ride rides. Perhaps we should have used more generic terms for the struct but at this point too much is using that name.

File: src/openrct2-ui/windows/EditorObjectiveOptions.cpp (hunk 1 above)
Comment (aramk-hub) [reply src/openrct2-ui/windows/EditorObjectiveOptions.cpp:138]: I see, I had a misunderstanding, thank you for the explanations though I think that helps me understand more of it all! Fixing this right now.

**Review by duncanspumpkin — APPROVED** [review]

**Review by ZehMatt — APPROVED** [review]
Tested all good, thanks.

**Pull request merged**